src/pages/index.js

```diff
@@ -258,27 +262,30 @@ const Index = ({ data: initialData }) => {
   useEffect(() => {
     setAdUnits(getAdUnits())
   }, [])
-
   // Only render ads if we are on the client side.
   const [shouldRenderAds, setShouldRenderAds] = useState(false)
   useEffect(() => {
     if (isClientSide()) {
       setShouldRenderAds(true)
     }
   }, [])
-
+  // FIXME: use UUID in state
+  const { app, user } = data || {}
+  const [userGobalId] = useState(get(user, 'id', ''))
+  // log tab count when user first visits
+  useEffect(() => {
+    if (userGobalId && tabId) {
+      console.log('fired an update', userGobalId, tabId)
```

Comment:
@kmjennison ok so i've been playing around with firing the tab logger from index.js.  Here's the problem, when you click on account settings, then navigate back to the main page, this effect fires again, even though the value of the variables hasn't changed.  I'm guessing this is because the component is unmounting / remounting.  My next thought is to put this in the myApp component (where I had to store the tab uuid so it didn't regenerate), but the problem is that we moved loading the userId to an hoc and we don't have access to it in myApp.  Does navigating to settings then returning back to the tab page count as a 'new tab'?  If we can't change the state on the frontend then perhaps we could add a check in logTab.js on the backend to query for userId,tabId on the UserTabsLogModel and only update if it doesn't exist?

Reply:
@amaliwanag2 It's all good if we log the tab again in that case. That's the current behavior. In theory, we would show new ads, too.

Reply:
To be more precise, it's a "view of the new tab dashboard", not necessarily an opening of a browser tab.